### PR TITLE
fix: disable save button when widgets are invalid

### PR DESCRIPTION
**Pull request by yuda110**
### To Reviewers
- [ ] Skip (`style`, `chore` ONLY)
- [x] Not that difficult

### Type of Change
- [ ] New feature
- [x] Bug fixes
- [x] Feature improvement
- [ ] Refactor
- [ ] Others (performance improvement, refactoring, CI/CD, etc.)

### Checklist
- [x] `Error / Warning / Lint / Type`

### Description
see commit messages

### CODE REVIEW

File: src/services/dashboards/dashboard-detail/modules/DashboardWidgetContainer.vue [hunk 1]
```diff
@@ -132,7 +132,7 @@ export default defineComponent<Props>({
             if (state.initiatedWidgetMap[target.id]) return;
             if (isIntersecting) {
                 const targetWidgetRef: WidgetComponent|null = state.widgetRef.find((d) => d?.$el?.id === target.id);
-                if (typeof targetWidgetRef?.initWidget === 'function') {
+                if (typeof targetWidgetRef?.initWidget === 'function' && dashboardDetailValidationState.widgetValidMap[target.id] !== false) {
```
Comment (yuda110) [inline src/services/dashboards/dashboard-detail/modules/DashboardWidgetContainer.vue:135]: description: Do not initWidget which is invalid

File: src/services/dashboards/dashboard-customize/modules/DashboardCustomizeSidebar.vue [hunk 2]
```diff
@@ -54,6 +54,7 @@
                     {{ $t('DASHBOARDS.CUSTOMIZE.CANCEL') }}
                 </p-button>
                 <p-button style-type="primary"
+                          :disabled="!dashboardDetailValidationState.isWidgetLayoutValid"
```
Comment (yuda110) [inline src/services/dashboards/dashboard-customize/modules/DashboardCustomizeSidebar.vue:57]: description: disable save button when widgets are invalid

File: src/services/dashboards/dashboard-detail/modules/DashboardWidgetContainer.vue (hunk 1 above)
Comment (minseolee) [inline src/services/dashboards/dashboard-detail/modules/DashboardWidgetContainer.vue:135]:
```suggestion
                if (typeof targetWidgetRef?.initWidget === 'function' && dashboardDetailValidationState.widgetValidMap[target.id] === true) {
```
Looks like `widgetValidMap` could be `undefined`. How about this ?!

Comment (yuda110) [reply src/services/dashboards/dashboard-detail/modules/DashboardWidgetContainer.vue:135]: If the widget is valid, it may not be in the `widgetValidMap`..

Comment (minseolee) [reply src/services/dashboards/dashboard-detail/modules/DashboardWidgetContainer.vue:135]: Oh .. got it

**Review by song2kim — APPROVED** [review]
LGTM~

**Review by sulmoJ — APPROVED** [review]
LGTM!!

**Pull request merged**